animation_nodes/nodes/mesh/set_vertex_weights.py

```diff
@@ -0,0 +1,157 @@
+        elif self.groupIdentifierType == "NAME":
+            self.newInput("Text", "Name", "groupName")
+
+        if self.mode == "INDEX":
+            self.newInput(VectorizedSocket("Integer", "useIndexList",
+                ("Index", "index"), ("Indices", "indices")))
+            self.newInput(VectorizedSocket("Float", "useWeightsList",
+                ("Weight", "weight"), ("Weights", "weights")))
+        elif self.mode == "ALL":
+            self.newInput(VectorizedSocket("Float", "useWeightsList",
+                ("Weight", "weight"), ("Weights", "weights")))
+        
+        self.newOutput("Object", "Object", "object")
+
+    def draw(self, layout):
+        layout.prop(self, "mode", text = "")
+
+    def drawAdvanced(self, layout):
+        layout.prop(self, "groupIdentifierType", text = "Type")
+
+    def getExecutionFunctionName(self):
+        if self.mode == "INDEX":
+            if self.useIndexList:
+                if self.useWeightsList:
+                    return "execute_IndicesList"
+                else:
+                    return "execute_IndicesSingle"
+            else:
+                return "execute_Index"
+        elif self.mode == "ALL":
+            if self.useWeightsList:
+                return "execute_AllList"
+            else:
+                return "execute_AllSingle"
+
+    def execute_Index(self, object, identifier, index, weight):
+        if object is None: return
+        if object.type != "MESH": 
+            self.raiseErrorMessage(noMeshMessage)
+        if object.mode == "EDIT":
+            self.raiseErrorMessage(editMeshMessage)
+
+        vertexGroup = self.getVertexGroup(object, identifier)
+        if vertexGroup is None:
+            self.raiseErrorMessage(groupNotFoundMessage)
+
+        vertexGroup.add([index], weight, "REPLACE")
+        object.data.update()    
+        return object
+
+    def execute_IndicesSingle(self, object, identifier, indices, weight):
+        if object is None: return
+        if object.type != "MESH": 
+            self.raiseErrorMessage(noMeshMessage)
+        if object.mode == "EDIT":
+            self.raiseErrorMessage(editMeshMessage)
+
+        vertexGroup = self.getVertexGroup(object, identifier)
+        if vertexGroup is None:
+            self.raiseErrorMessage(groupNotFoundMessage)
+
+        vertexGroup.add(indices, weight, "REPLACE")
+        object.data.update()    
+        return object
+
+    def execute_IndicesList(self, object, identifier, indices, weights):
+        if object is None: return
+        if object.type != "MESH": 
+            self.raiseErrorMessage(noMeshMessage)
+        if object.mode == "EDIT":
+            self.raiseErrorMessage(editMeshMessage)
+
+        vertexGroup = self.getVertexGroup(object, identifier)
+        if vertexGroup is None:
+            self.raiseErrorMessage(groupNotFoundMessage)
+
+        for i, index in enumerate(indices):
+            vertexGroup.add([index], weights[i], "REPLACE")
```

Comment:
Use virtualDoubleList here as well.

Reply:
Okay. I'll add the virtualDoubleList here.